pkg/utils/get_ipfamily.go

```diff
@@ -26,3 +33,88 @@ func GetIPFamilyFromCidr(cidr string) string {
 
 	return addrType
 }
+
+func ipFamilyFromCIDRStrings(cidrs []string) (string, error) {
+	if len(cidrs) > 2 {
+		return InvalidIPFamily, errors.New("too many CIDRs specified")
+	}
+	var foundIPv4 bool
+	var foundIPv6 bool
+	for _, cidr := range cidrs {
+		cidrType := GetIPFamilyFromCidr(cidr)
+		if cidrType == IPv4IpFamily {
+			foundIPv4 = true
+		} else if cidrType == IPv6IpFamily {
+			foundIPv6 = true
+		}
+	}
+	switch {
+	case foundIPv4 && foundIPv6:
+		return DualStackIPFamily, nil
+	case foundIPv4:
+		return IPv4IpFamily, nil
+	case foundIPv6:
+		return IPv6IpFamily, nil
+	default:
+		return InvalidIPFamily, errors.New("Invalid IP Family")
+	}
+}
+
+// GetClusterIPFamily returns a cluster IPFamily from the configuration provided.
+func GetClusterIPFamily(c *capi.Cluster) (string, error) {
+	var podCIDRs, serviceCIDRs []string
+	var podsIPFamily, servicesIPFamily string
+	var err error
+
+	if c.Spec.ClusterNetwork != nil {
+		if c.Spec.ClusterNetwork.Pods != nil {
+			podCIDRs = c.Spec.ClusterNetwork.Pods.CIDRBlocks
+		}
+		if c.Spec.ClusterNetwork.Services != nil {
+			serviceCIDRs = c.Spec.ClusterNetwork.Services.CIDRBlocks
+		}
+	}
+	if len(podCIDRs) == 0 && len(serviceCIDRs) == 0 {
```

Comment:
Could you also add some comments to explain this tool function?

Some comments can help us to pick up quickly in future